app/Http/Controllers/Api/PostController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use App\Http\Requests\PostRequest;
+use App\Http\Controllers\Controller;
+use App\Http\Resources\PostResource;
+use Illuminate\Support\Facades\Validator;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index(): JsonResponse
+    {
+        $posts = Post::with('creator','likes', 'comments', 'shares')->withCount('likes', 'comments', 'shares')->latest()->paginate(10);
```

Comment:
Please add size attribute  and page to emulate the pagination

Reply:
Okay